specification/sbi_atee.adoc

```diff
@@ -235,24 +235,29 @@ reclaiming the memory for use by non-confidential workloads requires an explicit
 `sbi_tee_host_reclaim_pages()`.
 
 ==== Operational model for the TEE Guest Extension
-This interface is used by TVMs to communicate with TSM. Presently, this extension only allows guests
-to define memory regions for shared-pages and MMIO regions.
-
-===== TVM-defined memory regions
-TVMs can determine the physical address location for mapping of non-confidential regions at runtime,
-and communicate the decision host about TVM-established shared-pages and MMIO regions by calling
-`sbi_tee_guest_add_memory_region()`. This results in an exit to the host, and it can retrieve the
-information by checking the exit code from the TVM and examining the shared-memory region for the
-TVM VCPU. The expectation is that the host will service a subsequent page-fault that results from
-a TVM-access to the non-confidential region.
-
-===== TVM-driven confidential/non-confidential memory conversion
+This interface is used by TVMs to communicate with TSM. Presently, this extension allows guests
+to define memory regions for MMIO emulation by host, share pages with the host and control interrupt
+injection by host.
+
+===== TVM-defined MMIO regions
+TVM can register the physical address location as a non-confidential MMIO region at runtime to be emulated by the host.
+This is done by calling `sbi_tee_guest_add_mmio_region()`. This results in an exit to the host, and it can retrieve the
+information by checking the exit code from the TVM and examining the NACL shared memory region. The expectation
+is that the host will service a subsequent page-fault that results from a TVM-access to the non-confidential region.
+
+===== TVM-defined Shared memory regions
 TVMs can choose to yield access to confidential memory at runtime and request shared (non-confidential) memory.
-The TVM must communicate it's request to the host to convert confidential to non-confidential and vice-versa 
-explicitly via the `sbi_tee_guest_(un)share_memory_region`. This request results in an exit to the TSM which 
-enforces the security properties on the mapping and exits to the VMM host to enforce TLB invalidation. 
-The expectation is that the host will service these requests and handle subsequent page-faults to allow the TVM 
-to access required confidential or non-confidential memory per the TVMs request.
+The TVM must communicate its request to the host to convert confidential to non-confidential and vice-versa 
+explicitly via the `sbi_tee_guest_share_memory_region()` and `sbi_tee_guest_unshare_memory_region()`. This request
+results in an exit to the TSM which enforces the security properties on the mapping and exits to the VMM host to 
+enforce TLB invalidation. The host must complete a TVM TLB invalidation sequence, initiated by `tee_host_tvm_initiate_fence()`,
```

Comment:
just noting there that this procedure is going to change such that the host must explicit make a `remove_pages()` call in order to support sharing/unsharing of hugepage-backed memory

Reply:
Noted.